service/operator/framework.go

```
@@ -42,6 +43,18 @@ func newFramework(config Config) (*framework.Framework, error) {
 		}
 	}
 
+	var randomkeysSearcher *randomkeys.Searcher
+	{
+		c := randomkeys.DefaultConfig()
+		c.K8sClient = config.K8sClient
+		c.Logger = config.Logger
+
+		randomkeysSearcher, err = randomkeys.NewSearcher(c)
+		if err != nil {
+			return nil, microerror.Maskf(err, "randomkeys.NewSearcher")
```

Here you can just mask the error. This is so we have the full call stack when it gets logged.

---

Isn't `microerror.Maskf` already doing that by providing both the call stack and an annotation?

---

It still works but AFAIK the annotation won't be logged. We usually use `Maskf` when creating an error. If you're returning it you can just use `Mask`.

---

Site node: annotation will be logged.

---

Good to know, thanks!